code/modules/projectiles/projectile/special/rocket.dm

```diff
@@ -70,6 +70,30 @@
 	else
 		explosion(target, light_impact_range = 2, flame_range = 3, flash_range = 4)
 
+/// PM9 solid fuel canister
+/obj/projectile/bullet/a84mm/solidfuel
+	name ="\improper Solid fuel canister"
+	desc = "Flamin' hot!"
+	icon_state = "solidfuel_syndie"
+	damage = 20 //still hurts to get smacked by a flaming canister at high speeds
+	dismemberment = 0
+	embedding = null
+	shrapnel_type = null
+	var/fire_stacks = 6
+
+/obj/projectile/bullet/a84mm/solidfuel/on_hit(atom/target, blocked = 0)
+	if(iscarbon(target))
+		if(isliving(target))
```

Comment:
carbon is living (/mob/living/carbon) isliving doesn't check if the target is living but if the target is /mob/living.
you probably want to drop the isliving check completely.